### ISSUE TITLE
QtValueError when opening URL with invalid characters

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#3633]
Report by @tjanson:

>  Clicked on a link in a PDF document (in Preview -- Mac OS High Sierra). The link is typeset as http://merkur.informatik.rwth-aachen.de/bscw/bscw.cgi/d3493213/thesis.pdf but actually points to http://aachen.de/bscw/bscw.cgi/d3493213/thesis.pdf (which is likely not a valid resource).

```
16:10:46 DEBUG    init       app:open_url:321 About to open URL:
16:10:46 ERROR    misc       crashsignal:exception_hook:216 Uncaught exception
Traceback (most recent call last):
  File "qutebrowser/app.py", line 836, in event
  File "qutebrowser/app.py", line 322, in open_url
  File "qutebrowser/mainwindow/tabbedbrowser.py", line 432, in tabopen
  File "qutebrowser/utils/qtutils.py", line 139, in ensure_valid
qutebrowser.utils.qtutils.QtValueError: PyQt5.QtCore.QUrl('') is not valid: Invalid hostname (contains invalid characters); source was "http://merkur.informatik.rwth-"; scheme = "http", host = ""
```

https://crashes.qutebrowser.org/view/d8177689

I think this is coming via macOS' OpenEvent but never checked for validity anywhere? I'm guessing the `-` in the PDF is probably some non-ASCII dash. cc @residentsummer

**Comment by tjanson** [comment]
> I'm guessing the - in the PDF is probably some non-ASCII dash.

At least when I copy-and-paste it from the document, it seems to be a regular hyphen (U+002D `-`). You can have a look yourself, the malformed links are at the end of [this page](https://github.com/qutebrowser/qutebrowser/files/1754729/las2peer.-.A.Primer.dragged.pdf).

**Comment by residentsummer** [comment]
I see it as a typesetting (or Preview's) problem. Placing the cursor over different parts of the link reveals that viewer threats it as two separate links (pointer is not visible on my screenshots :( ).

First half:
<img width="518" alt="screen shot 2018-02-25 at 12 06 56" src="https://user-images.githubusercontent.com/813112/36639813-649ffea6-1a25-11e8-9b2a-0c221ec31e58.png">

Second half:
<img width="553" alt="screen shot 2018-02-25 at 12 07 07" src="https://user-images.githubusercontent.com/813112/36639823-8914cac8-1a25-11e8-8615-33368e390d17.png">

Selecting that text with a mouse yields a link with a whitespace after the dash. So I guess Qt is rightfully declares provided URL as invalid (dashes are not allowed in TLD part, if my interpretation of RFC is correct).

**Comment by tjanson** [comment]
Yes, I think it's undisputed that the links are malformed; the problem is
that it crashes qutebrowser rather than yielding an error message.

>

**Comment by residentsummer** [comment]
@tjanson it definitely shouldn't crash :) Thanks for reporting!

**Comment by The-Compiler** [comment]
Fixed in #3637 - thanks @residentsummer and @tjanson!
